### PR TITLE
EZR-4556: Add Files to Leads

### PR DESCRIPTION
**Pull request by fatebinder**
Added support to add/remove Files for Leads

### CODE REVIEW

File: src/Request/AddLeadFile.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    /**
+     * @var string
+     */
+    public $responseClass = Attachment::class;
+    /**
+     * @var string
+     */
+    protected $modelClass = RequestModel\AddLeadFile::class;
+    /**
+     * @var string[]
```
Comment (mortenhauberg) [inline src/Request/AddLeadFile.php]: Incorrect. Should be `array`.

File: src/Request/AddLeadFile.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+     */
+    public $options = ['header' => ['Content-Type' => 'multipart/form-data']];
+
+    /**
+     * @param RequestModel $model
```
Comment (mortenhauberg) [inline src/Request/AddLeadFile.php]: `RequestModel |null`

Comment (fatebinder) [reply src/Request/AddLeadFile.php]: May I ask why it should be |null also? We don't make the request without a Model

Comment (mortenhauberg) [reply src/Request/AddLeadFile.php]: Sure :) There's a `null` default value

Comment (mortenhauberg) [reply src/Request/AddLeadFile.php]: Which we don't handle.. If you pass a `null` value, it will break on line 42

Comment (mortenhauberg) [reply src/Request/AddLeadFile.php]: @fatebinder we still need to update the doc block. The parameter _can_ be either `RequestModel` or `null`

**Pull request merged**